src/v/archival/archival_policy.cc

```diff
@@ -26,17 +27,40 @@ namespace archival {
 using namespace std::chrono_literals;
 
 std::ostream& operator<<(std::ostream& s, const upload_candidate& c) {
-    s << "{ exposed_name: " << c.exposed_name
-      << ", starting_offset: " << c.starting_offset
-      << ", segment_file_name: " << c.source->reader().filename() << " }";
+    fmt::print(
+      s,
+      "{{segment: {}, exposed_name: {}, starting_offset:{},  final_offset:{}, "
+      "file_offset: {}, content_length: {}}}",
+      *c.source,
+      c.exposed_name,
+      c.starting_offset,
+      c.final_offset,
+      c.file_offset,
+      c.content_length);
     return s;
 }
 
 archival_policy::archival_policy(
-  model::ntp ntp, service_probe& svc_probe, ntp_level_probe& ntp_probe)
+  model::ntp ntp,
+  service_probe& svc_probe,
+  ntp_level_probe& ntp_probe,
+  std::optional<segment_time_limit> limit)
   : _ntp(std::move(ntp))
   , _svc_probe(svc_probe)
-  , _ntp_probe(ntp_probe) {}
+  , _ntp_probe(ntp_probe)
+  , _upload_limit(limit) {}
+
+bool archival_policy::upload_limit_reached() {
```

Comment:
`upload_deadline_reached`? `upload_limit` doesn't sound like like anything related to time...